Source/Scene/ModelExperimental/ModelExperimentalArticulationStage.js

```diff
@@ -0,0 +1,260 @@
+      return this._name;
+    },
+  },
+
+  /**
+   * The type of this articulation stage. This specifies which of the
+   * node's properties is modified by the stage's value.
+   *
+   * @memberof ModelExperimentalArticulationStage.prototype
+   * @type {ArticulationStageType}
+   * @readonly
+   *
+   * @private
+   */
+  type: {
+    get: function () {
+      return this._type;
+    },
+  },
+
+  /**
+   * The minimum value of this articulation stage.
+   *
+   * @memberof ModelExperimentalArticulationStage.prototype
+   * @type {Number}
+   * @readonly
+   *
+   * @private
+   */
+  minimumValue: {
+    get: function () {
+      return this._minimumValue;
+    },
+  },
+
+  /**
+   * The maximum value of this articulation stage.
+   *
+   * @memberof ModelExperimentalArticulationStage.prototype
+   * @type {Number}
+   * @readonly
+   *
+   * @private
+   */
+  maximumValue: {
+    get: function () {
+      return this._maximumValue;
+    },
+  },
+
+  /**
+   * The current value of this articulation stage.
+   *
+   * @memberof ModelExperimentalArticulationStage.prototype
+   * @type {Number}
+   *
+   * @private
+   */
+  currentValue: {
+    get: function () {
+      return this._currentValue;
+    },
+    set: function (value) {
+      //>>includeStart('debug', pragmas.debug);
+      Check.typeOf.number("value", value);
+      //>>includeEnd('debug');
+
+      value = CesiumMath.clamp(value, this.minimumValue, this.maximumValue);
+      if (
+        !CesiumMath.equalsEpsilon(
+          this._currentValue,
+          value,
+          articulationEpsilon
+        )
+      ) {
+        this._currentValue = value;
+        this.runtimeArticulation._dirty = true;
+      }
+    },
+  },
+});
+
+const scratchArticulationCartesian = new Cartesian3();
+const scratchArticulationRotation = new Matrix3();
+
+/**
+ * Modifies a Matrix4 by applying a transformation for a given value of a stage.
+ * Note this is different usage from the typical <code>result</code> parameter,
+ * in that the incoming value of <code>result</code> is meaningful. Various stages
+ * of an articulation can be multiplied together, so their transformations are
+ * all merged into a composite Matrix4 representing them all.
+ *
+ * @param {Matrix4} result The matrix to be modified.
+ * @returns {Matrix4} The transformed matrix as requested by the articulation stage.
+ *
+ * @private
+ */
+ModelExperimentalArticulationStage.prototype.applyStageToMatrix = function (
+  result
+) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("result", result);
+  //>>includeEnd('debug');
+
+  const type = this.type;
+  const value = this.currentValue;
+  const cartesian = scratchArticulationCartesian;
+  let rotation;
+  switch (type) {
+    case ArticulationStageType.XROTATE:
+      rotation = Matrix3.fromRotationX(
+        CesiumMath.toRadians(value),
+        scratchArticulationRotation
+      );
+      Matrix4.multiplyByMatrix3(result, rotation, result);
```

Comment:
I know there's been some discussion about this, but the style guide currently requires these be the (somewhat redundant) `result = Matrix4.multiplyByMatrix3()`